src/arithmetic.jl

```diff
@@ -18,63 +22,203 @@ for f in [:inv, :sqrt, :log, :log1p, :exp, :expm1, :atan, :cosh, :sinh]
-Base.max(x::MagOrRef, y::MagOrRef) = max!(zero(x), x, y)
-Base.minmax(x::MagOrRef, y::MagOrRef) = (min(x, y), max(x, y))
+### Arf
+function Base.sign(x::ArfOrRef)
+    isnan(x) && return Arf(NaN) # Follow Julia and return NaN
+    return Arf(sgn(x))
+end
 
-### Arf, Arb and Acb
+Base.abs(x::ArfOrRef) = abs!(zero(x), x)
+Base.:(-)(x::ArfOrRef) = neg!(zero(x), x)
 for (jf, af) in [(:+, :add!), (:-, :sub!), (:*, :mul!), (:/, :div!)]
-    @eval function Base.$jf(x::T, y::T) where {T<:Union{Arf,ArfRef,Arb,ArbRef,Acb,AcbRef}}
-        z = T(prec = max(precision(x), precision(y)))
+    @eval function Base.$jf(x::ArfOrRef, y::Union{ArfOrRef,_BitInteger})
+        z = Arf(prec = _precision((x, y)))
         $af(z, x, y)
-        z
+        return z
     end
 end
-function Base.:(-)(x::T) where {T<:Union{Arf,ArfRef,Arb,ArbRef,Acb,AcbRef}}
-    z = T(prec = precision(x))
-    neg!(z, x)
-    z
+function Base.:+(x::_BitInteger, y::ArfOrRef)
+    z = zero(y)
+    add!(z, y, x)
+    return z
+end
+function Base.:*(x::_BitInteger, y::ArfOrRef)
+    z = zero(y)
+    mul!(z, y, x)
+    return z
+end
+function Base.:/(x::_BitUnsigned, y::ArfOrRef)
+    z = zero(y)
+    ui_div!(z, x, y)
+    return z
+end
+function Base.:/(x::_BitSigned, y::ArfOrRef)
+    z = zero(y)
+    si_div!(z, x, y)
+    return z
+end
+
+function Base.sqrt(x::ArfOrRef)
+    y = zero(x)
+    sqrt!(y, x)
+    return y
+end
+function rsqrt(x::ArfOrRef)
+    y = zero(x)
+    rsqrt!(y, x)
+    return y
+end
+function root(x::ArfOrRef, k::Integer)
+    y = zero(x)
+    root!(y, x, convert(UInt, k))
+    return y
+end
+
+### Arb and Acb
+for (jf, af) in [(:+, :add!), (:-, :sub!), (:*, :mul!), (:/, :div!)]
+    @eval Base.$jf(x::ArbOrRef, y::Union{ArbOrRef,ArfOrRef,_BitInteger}) =
+        $af(Arb(prec = _precision((x, y))), x, y)
+    @eval Base.$jf(x::AcbOrRef, y::Union{AcbOrRef,ArbOrRef,_BitInteger}) =
+        $af(Acb(prec = _precision((x, y))), x, y)
+    if jf == :(+) || jf == :(*)
+        @eval Base.$jf(x::Union{ArfOrRef,_BitInteger}, y::ArbOrRef) =
+            $af(Arb(prec = _precision((x, y))), y, x)
+        @eval Base.$jf(x::Union{ArbOrRef,_BitInteger}, y::AcbOrRef) =
+            $af(Acb(prec = _precision((x, y))), y, x)
+    end
 end
-function Base.:(^)(x::T, k::Integer) where {T<:Union{Arb,ArbRef,Acb,AcbRef}}
-    z = T(prec = precision(x))
-    pow!(z, x, convert(UInt, k))
-    z
+
+Base.:(-)(x::Union{ArbOrRef,AcbOrRef}) = neg!(zero(x), x)
+Base.abs(x::ArbOrRef) = abs!(zero(x), x)
+Base.:(/)(x::_BitUnsigned, y::ArbOrRef) = ui_div!(zero(y), x, y)
+
+Base.:(^)(x::ArbOrRef, y::ArbOrRef) = pow!(Arb(prec = _precision((x, y))), x, y)
+function Base.:(^)(x::ArbOrRef, y::_BitInteger)
+    z = zero(x)
+    x, n = (y >= 0 ? (x, y) : (inv!(z, x), -y))
+    return pow!(z, x, convert(UInt, n))
 end
+Base.:(^)(x::AcbOrRef, y::Union{AcbOrRef,ArbOrRef,_BitInteger}) =
+    pow!(Acb(prec = _precision((x, y))), x, y)
+
+Base.literal_pow(::typeof(^), x::Union{ArbOrRef,AcbOrRef}, ::Val{2}) = sqr(x)
```

Comment:
let's go full in ;)
```suggestion
Base.literal_pow(::typeof(^), x::Union{ArbOrRef,AcbOrRef}, ::Val{2}) = sqr(x)
Base.literal_pow(::typeof(^), x::Union{ArbOrRef,AcbOrRef}, ::Val{1}) = copy(x)
Base.literal_pow(::typeof(^), x::Union{ArbOrRef,AcbOrRef}, ::Val{0}) = one(x)
# Base.literal_pow(::typeof(^), x::Union{ArbOrRef,AcbOrRef}, ::Val{-1}) = inv(x) : implemented in Base.intfuncs.jl
Base.literal_pow(::typeof(^), x::Union{ArbOrRef,AcbOrRef}, ::Val{-2}) = (res = sqr(x); inv!(res, res))
```

Reply:
Done! It's a bit complicated to debug these things since `@which, @less, @enter` all makes it not use `literal_pow` anymore. But with some debug printing and performance comparisons I'm fairly certain they are actually used.